src/unstract/sdk/llm.py

```diff
@@ -45,18 +46,22 @@ def __init__(
         """
         self._tool = tool
         self._adapter_instance_id = adapter_instance_id
-        self._llm_instance: LlamaIndexLLM = self._get_llm(self._adapter_instance_id)
-
-        self._usage_kwargs = usage_kwargs.copy()
-        self._usage_kwargs["adapter_instance_id"] = adapter_instance_id
-        platform_api_key = self._tool.get_env_or_die(ToolEnv.PLATFORM_API_KEY)
-        CallbackManager.set_callback_manager(
-            platform_api_key=platform_api_key,
-            model=self._llm_instance,
-            kwargs=self._usage_kwargs,
-        )
+        self._llm_instance: LlamaIndexLLM = None
```

Comment:
@gaya3-zipstack If I want to use the LlamaIndex LLM instance directly for some reason, then is accessing like `llm._llm_instance` enough to do this from hereon?

Reply:
Yes. But _llm_instance is a protected field right? Should I add a accessor for that? I would prefer not to. But Python will anyway allow access to the variable I guess!

Reply:
@gaya3-zipstack Yes let's keep it protected, because normally it need not be accessed directly as you mentioned.

However there could be some Enterprise workflows (e.g. multi-doc chat) or a new LlamaIndex feature that we might need on a use case basis for which this would be required, that's all.